### ISSUE TITLE
[Epic] Wallet Connect MVP

### ISSUE THREAD

**Issue by alexjba** [status-im/status-desktop#16312]
# Content

| Story | Priority | Status |  Details |
|--------|--------|--------|--------|
| [Add SIWE to sign-in via Wallet Connect](https://www.notion.so/Add-SIWE-to-sign-in-via-Wallet-Connect-c3e910df4dd743f582be8e0ffd413b0b?pvs=4) | High | ✍️  User Story Review | The connection establishment flow with dApps implementing the sign in is fragile without implementing SIWE. SIWE allows the user to easily authenticate |
| [Ensure Wallet Connect complies with EIP and WC standards in terms of API implementations](https://www.notion.so/Ensure-Wallet-Connect-complies-with-EIP-and-WC-standards-in-terms-of-API-implementations-2fad01c858d449b294bb213b4d7751e0?pvs=4) | High | ✍️ User Story Review | Analyse and implement the required APIs based on standards and best practices |
| [Fix transaction issues](https://www.notion.so/Fix-transaction-issues-9396d11c303d4486b5e728263b4e47f4?pvs=4) | High | ⌨️  Dev ready | Fix the bugs preventing the user to complete some transactions |
| [Sign and transaction request expiration](https://www.notion.so/Sign-and-transaction-request-expiration-7533275831f3407398ccf8b0603bacbe?pvs=4) | High | ✍️ User Story Review | Don't allow sign or transaction actions if the request has expired |
| [Session expiration](https://www.notion.so/Session-expiration-215cdb75dae540a68098003df8c1f5e6?pvs=4) | Low | ✍️ User Story Review | Try to extend a session before expiry. Otherwise delete the session |
| [Proposal expiration](https://www.notion.so/Proposal-expiration-9132158ab3754125bcfb5dc67fc08097?pvs=4) | Low | ✍️ User Story Review | Avoid accepting expired connections |
| [Metrics for Wallet Connect](https://www.notion.so/Metrics-for-Wallet-Connect-d3d1e60e7ea24479bd35cbb11219540c?pvs=4) | Low | ✍️ User Story Review | Sending analytics messages from critical flows |
| Release preparations | High |  ⌨️  Dev ready | Prepare the DB indexes and Enable WC by default |

> *High* priority are the tasks needed for the release
> *Low* priority are nice to have in the release

# Details

### [Add SIWE to sign-in via Wallet Connect](https://www.notion.so/Add-SIWE-to-sign-in-via-Wallet-Connect-c3e910df4dd743f582be8e0ffd413b0b?pvs=4)

UI tasks: 
- [ ] #16317
- [ ] #16319
- [ ] #16320

Integration tasks:
- [x] https://github.com/status-im/status-desktop/issues/14996

Bugs:
- [x] https://github.com/status-im/status-desktop/issues/15704
- [x] #16113

<details><summary> Abstract </summary>
<p>

 Sign-In with Ethereum describes how Ethereum accounts authenticate with off-chain services by signing a standard message format parameterized by scope, session details, and security mechanisms (e.g., a nonce). The goals of this specification are to provide a self-custodied alternative to centralized identity providers, improve interoperability across off-chain services for Ethereum-based authentication, and provide wallet vendors a consistent machine-readable message format to achieve improved user experiences and consent management.

Replacing centralised standards: Replaces eg entering a phone number and receiving a code via text to prove you own/ have access to that phone number. Connecting is similar to entering the phone number. Signing is similar to entering the confirmation code you received on that number. In the sign in case, the way to do it is by generating a cryptographic nonce for you to sign with your private key. That signature proves without a doubt that you are indeed the owner of your wallet. 

</p>
</details> 

<details><summary>Notes</summary>
<p>

 SIWE enables users to easily sign in to dApps by authenticating with their wallet.

The plan is to support SIWE standard message format as specified in ERC-4361. When the dApp prompts a user to sign a message that follows the SIWE format, we should parse the message and give the user this human readable dialog to sign into the dApp.

Not just for Ethereum - work with L2s and other EVM-compatible chains
Spruce leads it’s development  (🔗 [article from Spruce](https://blog.spruceid.com/siwe-evm/))
SSX enables developers to configure the fields of their SIWE message using the `siweConfig` option. This option allows you to overwrite the fields found in the SIWE message

</p>
</details> 

<details><summary>Useful links</summary>
<p>

🔗 [Figma](https://www.figma.com/design/HrmZp1y4S77QJezRFRl6ku/dApp-Interactions---Milestone-1?node-id=3967-273224&t=0iY08HSK6dT1WECK-4)
🔗 [ERC-4361: Sign-In with Ethereum ](https://eips.ethereum.org/EIPS/eip-4361)
🔗 [SIWE (MetaMask article)](https://support.metamask.io/privacy-and-security/why-am-i-being-asked-to-sign-in-sign-in-with-ethereum/)
🔗 [SIWE (MetaMask dev)](https://docs.metamask.io/wallet/how-to/sign-data/siwe/)
🔗 [ERC-4361: Sign-In with Ethereum](https://eips.ethereum.org/EIPS/eip-4361)
🔗 [Wallet connect One-click authentication](https://docs.walletconnect.com/walletkit/web/one-click-auth)

</p>
</details> 

#

### [Ensure Wallet Connect complies with EIP and WC standards in terms of API implementations](https://www.notion.so/Ensure-Wallet-Connect-complies-with-EIP-and-WC-standards-in-terms-of-API-implementations-2fad01c858d449b294bb213b4d7751e0?pvs=4)

UI tasks: 

Integration tasks:
- [x] https://github.com/status-im/status-desktop/issues/16180
- [ ] #16316
- [ ] Handle proposal expiry
- [ ] Handle sign/transaction expiry
- [x] https://github.com/status-im/status-desktop/issues/15916

Bugs:
- [x] https://github.com/status-im/status-desktop/issues/16020
- [x] https://github.com/status-im/status-desktop/issues/16114
- [ ] https://github.com/status-im/status-desktop/issues/15636

<details><summary>Useful links</summary>
<p>

https://docs.walletconnect.com/walletkit/best-practices

</p>
</details> 
# 

### [Fix transaction issues](https://www.notion.so/Fix-transaction-issues-9396d11c303d4486b5e728263b4e47f4?pvs=4)

Bugs:

- [x] https://github.com/status-im/status-desktop/issues/16096
- [x] https://github.com/status-im/status-desktop/issues/16063
- [x] https://github.com/status-im/status-desktop/issues/16117
- [x] https://github.com/status-im/status-desktop/issues/16004

# 

### [Sign and transaction request expiration](https://www.notion.so/Sign-and-transaction-request-expiration-7533275831f3407398ccf8b0603bacbe?pvs=4)

UI tasks: 
- [x] #16314

Integration tasks:
- [x] #16315

Bugs:
- [x] https://github.com/status-im/status-desktop/issues/16114

<details><summary>Notes</summary>
<p>

Notes for development

Request expiry parameter recently added by WC to allow dApps to extend the previously hardcoded 5min window:  🔗 [WC github feat: optional request expiry #1851](https://github.com/WalletConnect/walletconnect-monorepo/pull/1851)
Added an optional expiry parameter to session requests where a number between 300 & 604800 seconds (5 minutes min - 7 days max) can be specified to extend a request timeout

</p>
</details> 

<details><summary>Useful links</summary>
<p>

🔗 [Figma](https://www.figma.com/design/HrmZp1y4S77QJezRFRl6ku/dApp-Interactions---Milestone-1?node-id=3967-288216&m=dev)
</p>
</details> 

#

### Release preparations 

- [x] https://github.com/status-im/status-desktop/issues/15673
- [ ] https://github.com/status-im/status-desktop/issues/14975 |

**Comment by alexjba** [comment]
Blocked by https://github.com/status-im/status-desktop/issues/16391
